lib/ansible/config/manager.py

```diff
@@ -473,6 +474,16 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                     raise AnsibleOptionsError('Invalid type for configuration option %s: %s' %
                                               (to_native(_get_entry(plugin_type, plugin_name, config)), to_native(e)))
 
+            # ensure choice is valid
+            if value is not None and defs[config].get('choices'):
+                if value not in defs[config].get('choices'):
+                    choices = defs[config].get('choices')
+                    if isinstance(choices, Mapping):
+                        choices = ', '.join(choices.keys())
+                    else:
+                        choices = ', '.join(choices)
```

Comment:
Why not sort these as well? Would require adjusting `test_laps_password.py` as well, though.